code/modules/power/solar.dm

```diff
@@ -90,12 +103,16 @@ var/list/solars_list = list()
 
 /obj/machinery/power/solar/on_update_icon()
 	..()
-	overlays.Cut()
-	if(stat & BROKEN)
-		overlays += image('icons/obj/power.dmi', icon_state = "solar_panel-b", layer = ABOVE_HUMAN_LAYER)
-	else
-		overlays += image('icons/obj/power.dmi', icon_state = "solar_panel", layer = ABOVE_HUMAN_LAYER)
-		src.set_dir(angle2dir(adir))
+	if(!isnull(panel))
+		if(stat & BROKEN)
+			panel.icon_state = "solar_panel-b"
+		else
+			panel.icon_state = "solar_panel"
+
+		var/matrix/rot_matrix = matrix()
+		rot_matrix.Turn(adir)
```

Comment:
For reference, after the convo on Discord, it should be possible to calc the effective direction of the panels based on the overmap ship, sun direction and ship fore direction via:
````
arctan(sun) + (foredir - mapdir)
````
(thanks @NataKilar)